sync/sync_head.go

```diff
@@ -42,7 +42,8 @@ func (s *Syncer[H]) Head(ctx context.Context) (H, error) {
 	defer s.getter.Unlock()
 	netHead, err := s.getter.Head(ctx)
 	if err != nil {
-		return netHead, err
+		log.Errorw("requesting head from trusted peer, returning subjective head which may not be recent", "sbjHead", sbjHead.Height(), "err", err)
```

Comment:
My only concern is error log, not to make it super spammy and less frightening to users, we may want to make it Warn